pybamm/__init__.py

```diff
@@ -115,11 +125,11 @@ def version(formatted=False):
     FiniteVolumeDiscretisation,
     NodeToEdge,
 )
-from .discretisations.base_mesh import KNOWN_DOMAINS
-from .discretisations.base_mesh import BaseMesh, BaseSubmesh
-from .discretisations.finite_volume_meshes import (
-    FiniteVolumeMacroMesh,
-    FiniteVolumeSubmesh,
+from .discretisations.pybamm_meshes import KNOWN_DOMAINS
+from .discretisations.pybamm_meshes import (
+    PybammMesh,
+    PybammSubMesh,
+    Pybamm1DUniformSubMesh,
```

Comment:
Wasn't sure if we would be adding a separate FenicsMesh class but happy to change this to just Mesh and SubMesh etc. I have left a more detailed comment below about this but I think the Mesh should be independent of the spatial method because a Mesh contains the cell centre locations and the cell edge locations and this can be used to implement either finite difference, finite volume, or finite element methods: each just broadcasts variables onto the mesh differently and implements the gradient and divergence differently.

Reply:
Ok I see your point here, happy to leave as `Mesh`, `SubMesh` and `Uniform1DSubMesh` for now and look into again when the context of adding other meshes is clearer